Replace git phr with GitHub CLI

ogajduse opened the pull request:
## Summary

Replaces the custom `git phr` alias dependency with the standard GitHub CLI tool.

## Changes

- Updated `bump_packaging` to use `gh pr create` instead of `git phr`
- **Dependency documentation**: Added GitHub CLI installation instructions to README

## Benefits

- Removes dependency on custom git aliases that users need to configure manually
- Uses standard tooling available in distribution packages
- Addresses git phr issue mentioned in #336

ogajduse committed:
Replace git phr with GitHub CLI and add dependency documentation

  - Replace git phr alias with gh pr create in bump_packaging script
  - Change push behavior to always push before creating PR
  - Add GitHub CLI installation documentation to README
  - Addresses git phr dependency issue mentioned in #336

  This removes the dependency on custom git aliases and uses the standard
  GitHub CLI tool that's available in distribution packages.

ekohl commented on bump_packaging:
```diff
@@ -51,9 +51,8 @@ else
 fi
 
 git commit -m "Release $FULLVERSION"
+git push "$PACKAGING_GIT_REMOTE" "HEAD:$REMOTE_BRANCH"
```
See https://github.com/theforeman/theforeman-rel-eng/pull/337 for an expansion of the aliases I use. Your PR here effectively implements what @evgeni suggested about using `gh`.

ogajduse committed:
Implement deterministic GitHub CLI fork management

- Add GITHUB_USER setting with auto-detection via gh CLI
- Add PACKAGING_FORK_REMOTE setting for configurable fork remote name
- Fix git push/PR creation logic in bump_packaging:
  - Remove direct push to target branch when creating PRs
  - Use deterministic branch names: release-$PROJECT-$FULLVERSION
  - Explicitly manage fork creation with gh repo fork
  - Use configured GITHUB_USER and PACKAGING_FORK_REMOTE settings
- Document new settings in README
- Ensures predictable behavior and reusable fork setup

ogajduse commented:
I've pushed updates that address the issues identified in the review. Changes I've made:

- Added `GITHUB_USER` setting with auto-detection via `gh api user --jq .login` (configurable in settings.local)
- Added `PACKAGING_FORK_REMOTE` setting for configurable fork remote name
- Use deterministic branch naming: `release-$PROJECT-$FULLVERSION`
- Explicit fork creation with `gh repo fork --remote-name`

Benefits:

- PR creation now creates a fork-to-upstream PR
- Predictable remote names, branch names, and user identification
- Other scripts can leverage the same fork setup and settings

ogajduse commented:
I initially considered using just `gh pr create` without explicit fork management, but decided against it for several reasons:

- `gh pr create` creates forks and remotes with unpredictable names, making it impossible to reliably reference the fork setup in other scripts.
- The existing codebase uses `GIT_REMOTE=upstream` and `PACKAGING_GIT_REMOTE=origin` conventions. Auto-managed remotes could create inconsistent setups.
- Other scripts need predictable settings they can reference. When things break, explicit steps are much clearer than auto-detection "magic."

ogajduse committed:
Differentiate branches based on rpm/deb flavor

ogajduse committed:
Reuse FEATURE_BRANCH across the script

lhellebr commented on settings:
```diff
@@ -71,8 +71,10 @@ GIT_REMOTE="${GIT_REMOTE:-upstream}"
 GIT_DEVELOP_BRANCH=develop
 GIT_STABLE_BRANCH="${VERSION}-stable"
 GITHUB_NAMESPACE=theforeman
+GITHUB_USER="${GITHUB_USER:-$(gh api user --jq .login)}"
```
True. In that case, there's also an option to keep this API call but do it only when needed.

ogajduse replied:
addressed

ekohl commented on settings:
```diff
@@ -72,8 +72,10 @@ GIT_PROTOCOL="${GIT_PROTOCOL:-https}"
+GITHUB_USER="${GITHUB_USER:-$USER}"
 PACKAGING_DIR="$GIT_DIR/foreman-packaging"
 PACKAGING_GIT_REMOTE=origin
+PACKAGING_FORK_REMOTE="${PACKAGING_FORK_REMOTE:-origin}"
```
These both default to origin, but is that really what we want? You'll end up running `gh repo fork theforeman/foreman-packaging --remote-name "origin"` but there is already an origin. Doesn't it make more sense to default to the GH username?
```suggestion
PACKAGING_FORK_REMOTE="${PACKAGING_FORK_REMOTE:-${GITHUB_USER}}"
```

ogajduse committed:
Default fork remote to GitHub username

- Change PACKAGING_FORK_REMOTE default from 'origin' to GITHUB_USER
- Simplify git push to use --set-upstream with plain HEAD
- Update documentation to reflect actual default behavior

ogajduse committed:
Run  in non-interactive mode

ekohl commented on bump_packaging:
```diff
@@ -53,7 +54,16 @@ fi
 if [[ $PACKAGING_PR == true ]] ; then
-	git phr -m "Release $FULLVERSION" -b "$REMOTE_BRANCH"
+	# Ensure fork exists with configured remote name
+	GH_PROMPT_DISABLED=1 gh repo fork theforeman/foreman-packaging --remote-name "$PACKAGING_FORK_REMOTE"
```
This isn't silent for me:
```console
$ GH_PROMPT_DISABLED=1 gh repo fork theforeman/foreman-packaging --remote-name ekohl
! ekohl/foreman-packaging already exists
```
Redirecting stdout to `/dev/null` also doesn't do that, but I'd be hesitant to redirect stderr to `/dev/null`. I don't think there is a good way without a lot of hacking/manual coding so perhaps just accept it?

Only think that mitigates it a bit is to check if a git remote with the name `$PACKAGING_FORK_REMOTE` already exists.

ogajduse replied:
> Only think that mitigates it a bit is to check if a git remote with the name $PACKAGING_FORK_REMOTE already exists.

Yes, that's what I plan to implement in a follow-up PR.

ogajduse commented:
Thank you for your feedback. I'll follow up with a PR to fix the worktree functionality in this script, as well as I'll include a patch to silence the `gh repo fork` command output.
